unit_tests/kernels/UnitTestScalarMassElem.C

```diff
@@ -0,0 +1,102 @@
+{0.00028941835051033, 0.00086825505153098, 0.0026047651545929, 0.00086825505153098, 9.6472783503442e-05, 0.00028941835051033, 0.00086825505153098, 0.00028941835051033},
+{-0.0036180084493555, -0.0012060028164518, -0.0036180084493555, -0.010854025348067, -0.0012060028164518, -0.00040200093881728, -0.0012060028164518, -0.0036180084493555},
+{-0.0029631933450284, -0.00098773111500948, -0.00032924370500316, -0.00098773111500948, -0.0088895800350853, -0.0029631933450284, -0.00098773111500948, -0.0029631933450284},
+{0.00038043765414573, 0.0011413129624372, 0.00038043765414573, 0.00012681255138191, 0.0011413129624372, 0.0034239388873116, 0.0011413129624372, 0.00038043765414573},
+{-0.0003583555634414, -0.0010750666903242, -0.0032252000709726, -0.0010750666903242, -0.0010750666903242, -0.0032252000709726, -0.0096756002129179, -0.0032252000709726},
+{0.00034184546463258, 0.00011394848821086, 0.00034184546463258, 0.0010255363938977, 0.0010255363938977, 0.00034184546463258, 0.0010255363938977, 0.0030766091816932}
+  };
+
+
+  static constexpr double rhs[8] = {
+-0.00091589254110655, -0.0062063621624234, -0.0012984190416109, -0.0055026555161997, -0.005904879831756, -0.0015624935573955, -0.0068051842105984, -0.00142282082153};
+
+} // advection_diffusion
+} // hex8_golds
+} // anonymous namespace
+
+#endif
+
+/// Scalar advection/diffusion (will use mixture fraction as scalar)
+TEST_F(MixtureFractionKernelHex8Mesh, NGP_scalar_mass)
+{
+  // FIXME: only test on one core
+  if (stk::parallel_machine_size(MPI_COMM_WORLD) > 1) 
+    return;
+
+  fill_mesh_and_init_fields(true);
+
+  // Setup solution options for default advection kernel
+  solnOpts_.meshMotion_ = false;
+  solnOpts_.meshDeformation_ = false;
+  solnOpts_.externalMeshDeformation_ = false;
+
+  int numDof = 1;
+  unit_test_utils::HelperObjects helperObjs(bulk_, stk::topology::HEX_8, numDof, partVec_[0]);
+
+  sierra::nalu::TimeIntegrator timeIntegrator;
+  timeIntegrator.timeStepN_ = 0.1;
+  timeIntegrator.timeStepNm1_ = 0.1;
+  timeIntegrator.gamma1_ = 1.0;
+  timeIntegrator.gamma2_ = -1.0;
+  timeIntegrator.gamma3_ = 0.0;
+
+  helperObjs.realm.timeIntegrator_ = &timeIntegrator;
+
+  // Initialize the kernel
+  std::unique_ptr<sierra::nalu::Kernel> massKernel(
+    new sierra::nalu::ScalarMassElemKernel<sierra::nalu::AlgTraitsHex8>(
+     bulk_, solnOpts_, scalarQ_, helperObjs.assembleElemSolverAlg->dataNeededByKernels_, false));
+
+  // Register the kernel for execution
+  helperObjs.assembleElemSolverAlg->activeKernels_.push_back(massKernel.get());
+
+  // Populate LHS and RHS
+  helperObjs.assembleElemSolverAlg->execute();
+
+#ifndef KOKKOS_ENABLE_CUDA
+  EXPECT_EQ(helperObjs.linsys->lhs_.extent(0), 8u);
+  EXPECT_EQ(helperObjs.linsys->lhs_.extent(1), 8u);
+  EXPECT_EQ(helperObjs.linsys->rhs_.extent(0), 8u);
+
+//  std::cerr.precision(14);
```

Comment:
Do we need to keep around this commented section? Might be useful to add a helper method in `HelperObjects` that could print out the lhs/rhs on demand (could be used elsewhere).

Reply:
Yes I'll clean that up. I meant to do that already, and forgot.